lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,328 @@
+                            false));
+        } else {
+            boolean anyVCsMigrated = tacticalVcs.stream().anyMatch(vc -> vc.getMigrated() != null);
+            boolean allVCsMigrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+
+            if (anyVCsMigrated && !allVCsMigrated) {
+                LOGGER.warn(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Not all VCs are migrated for this user (%s).",
+                                        ReportUserIdentityItem.getUserHash(userId))));
+            }
+
+            reportUserIdentityItems.add(
+                    new ReportUserIdentityItem(
+                            userId,
+                            reportUserIdentityService
+                                    .getStrongestAttainedVotForCredentials(tacticalVcs)
+                                    .orElse(Vot.P0)
+                                    .name(),
+                            tacticalVcs.size(),
+                            reportUserIdentityService.getIdentityConstituent(tacticalVcs),
+                            allVCsMigrated));
+        }
+    }
+
+    private void aggregateAndPersistIdentities(List<ReportUserIdentityItem> items) {
+        LOGGER.info(LogHelper.buildLogMessage("Updating aggregate identities values."));
+        var reportSummaryItem =
+                reportSummaryScanDynamoDataStore.getItem(ScanDynamoDataStore.KEY_VALUE);
+        if (reportSummaryItem == null) {
+            reportSummaryItem =
+                    new ReportSummaryItem(ScanDynamoDataStore.KEY_VALUE, 0L, 0L, 0L, 0L);
+        }
+        List<ReportUserIdentityItem> totalP2Identities =
+                items.stream().filter(ui -> Vot.P2.name().equals(ui.getIdentity())).toList();
+        reportSummaryItem.setTotalP2(reportSummaryItem.getTotalP2() + totalP2Identities.size());
+        reportSummaryItem.setTotalP2Migrated(
+                reportSummaryItem.getTotalP2Migrated()
+                        + totalP2Identities.stream()
+                                .filter(item -> (Boolean.TRUE.equals(item.getMigrated())))
+                                .toList()
+                                .size());
+        reportSummaryItem.setTotalP1(
+                reportSummaryItem.getTotalP1()
+                        + items.stream()
+                                .filter(ui -> Vot.P1.name().equals(ui.getIdentity()))
+                                .toList()
+                                .size());
+        reportSummaryItem.setTotalP0(
+                reportSummaryItem.getTotalP0()
+                        + items.stream()
+                                .filter(ui -> Vot.P0.name().equals(ui.getIdentity()))
+                                .toList()
+                                .size());
+        reportSummaryScanDynamoDataStore.update(reportSummaryItem);
```

Comment:
One thing I'm concerned about here is whether we'll get any race-conditions due to eventual consistency. Is it possible that we will update the count and then the next page will try and update the count before the previous one has finished?

The best option is probably to turn this update into an update expression, which would do something like:
`SET total_p0 = total_p0 + :newp0` (etc.)

(Or go back to aggregating the user report table in a separate scan)

Reply:
Another scan will be same additional scan as in step-2. 

Also with current approach there is chance of corrupt summary data...... i.e if user run the the step-2 again with same lastEvaluatedKey then while doing step-2 we'll currupt summary data.

Reply:
What's your suggestion?

Reply:
if table scan is quick then we can separate the step-3 from step-2.